test/integration/apiserver/apply/apply_test.go

```diff
@@ -555,3 +557,251 @@ func TestApplyRemoveContainerPort(t *testing.T) {
+// patch object and the live object will error
+func TestApplyFailsWithVersionMismatch(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	obj := []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"}
+		},
+		"spec": {
+			"replicas": 3,
+			"selector": {
+				"matchLabels": {
+					 "app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`)
+
+	_, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body(obj).Do().Get()
+	if err != nil {
+		t.Fatalf("Failed to create object using Apply patch: %v", err)
+	}
+
+	obj = []byte(`{
+		"apiVersion": "extensions/v1beta",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"}
+		},
+		"spec": {
+			"replicas": 100,
+			"selector": {
+				"matchLabels": {
+					"app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`)
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body([]byte(obj)).Do().Get()
+	if err == nil {
+		t.Fatalf("Expecting to get version mismatch when applying object")
+	}
+	status, ok := err.(*errors.StatusError)
+	if !ok {
+		t.Fatalf("Expecting to get version mismatch as API error")
+	}
+	if status.Status().Code != http.StatusBadRequest {
+		t.Fatalf("expected status code to be %d but was %d", http.StatusBadRequest, status.Status().Code)
+	}
+}
+
+// TestApplyConvertsManagedFieldsVersion checks that the apply
+// converts the API group-version in the field manager
+func TestApplyConvertsManagedFieldsVersion(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	obj := []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"},
+			"managedFields": [
+				{
+					"manager": "create_test",
+					"operation": "Apply",
+					"apiVersion": "extensions/v1beta1",
+					"fields": {
+						"f:metadata": {
+							"f:labels": {
+								"f:app": {}
```

Comment:
maybe add another label here, that is not going to be acquired by the apply operation, and make sure that it's still owned below by "create_test"?

Reply:
Ah it won't be owned by `create_test` because we're creating the object w/ `apply_test`, so `apply_test` would take ownership of everything in the new object and would remove the new label from the field manager for the new object

Reply:
But that's the thing, `apply_test` is not going to grab everything from managedFields, it's only going to own the fields that it sets?

Reply:
Oh, though the field doesn't exist, so I don't know if it's still going to be owned ... Maybe you'll have to run one more operation somehow

Reply:
Ah okay, so it looks like the managed fields aren't decoded from the patch/new object for a server side apply, and they're only declarable, decoded, and set w/ an Update for the non-server-side apply case:

https://github.com/kubernetes/apiserver/blob/master/pkg/endpoints/handlers/fieldmanager/fieldmanager.go#L93-L105

Reply:
Yeah, we really don't want people to apply with this field set ...

So here's what I suggest you do:
- Run a `Create()` with a basic object AND `managedFields` set to some values, probably the same thing you have in your apply.
- Run a Apply to see that you can capture some of the values, even if they come from different group/versions.

I think you have most of the pieces, you just have to re-organize them

Does it make sense?